src/main/scala/cromwell/engine/db/slick/ExecutionComponent.scala

```diff
@@ -25,21 +25,43 @@ trait ExecutionComponent {
     override def * = (workflowExecutionId, callFqn, status, executionId.?) <>
       (Execution.tupled, Execution.unapply)
 
-    def workflow = foreignKey(
+    def workflowExecution = foreignKey(
       "FK_EXECUTION_WORKFLOW_EXECUTION_ID", workflowExecutionId, workflowExecutions)(_.workflowExecutionId)
 
     def uniqueKey = index("UK_EX_WORKFLOW_EXECUTION_ID",
       (workflowExecutionId, callFqn), unique = true)
   }
 
-  val executions = TableQuery[Executions]
+  protected val executions = TableQuery[Executions]
 
   val executionsAutoInc = executions returning executions.
     map(_.executionId) into ((a, id) => a.copy(executionId = Some(id)))
 
-  val executionByID = Compiled(
-    (id: Rep[Int]) => for {
+  val executionCallFqnsAndStatusesByWorkflowExecutionId = Compiled(
+    (workflowExecutionId: Rep[Int]) => for {
       execution <- executions
-      if execution.executionId === id
+      if execution.workflowExecutionId === workflowExecutionId
+    } yield (execution.callFqn, execution.status))
+
+  val executionsByWorkflowExecutionUuidAndCallFqn = Compiled(
+    (workflowExecutionUuid: Rep[String], callFqn: Rep[String]) => for {
+      execution <- executions
+      if execution.callFqn === callFqn
+      workflowExecution <- workflowExecutions
+      if workflowExecution.workflowExecutionUuid === workflowExecutionUuid
     } yield execution)
+
+  val executionStatusesByExecutionId = Compiled(
+    (executionId: Rep[Int]) => for {
+      execution <- executions
+      if execution.executionId === executionId
+    } yield execution.status)
+
+  // see workflowExecutionsByStatuses
+  def executionStatusesByWorkflowExecutionIdAndCallFqns(workflowExecutionId: Int, callFqns: Traversable[String]) = {
```

Comment:
My offer for any and all naming suggestions stands. ;) I don't have anything better off the top of my head.

The projections are weird. Interestingly in Slick, one doesn't even need the `case class` objects, you can map everything to just [tuples](http://slick.typesafe.com/doc/3.0.0/schemas.html#table-rows). I think the [mapped tables](http://slick.typesafe.com/doc/3.0.0/schemas.html#mapped-tables) just look neater.

To perform any selects or updates, we need to project to a set of column names to and a filter, to let slick know which cells in the database table to return or update. So these query result objects are a combo of:

1. A set of columns: `(FIRST_NAME, LAST_NAME, AGE)`
2. A row filter: `(AGE > 10 && LAST_NAME LIKE 'S%')`

Theses queries can actually then be used for `SELECT`, `UPDATE`, plus `DELETE` on whole rows matching the query. One can insert too, but those actions often look different, and we using auto-incrementing a lot.

See `WorkflowExecutionComponent.workflowExecutionStatusesByWorkflowExecutionUuid` as an example. The one `Compiled[Query]` is used for:

- `SELECT`/`.result` in `SlickDataAccess.getWorkflowState`
- `UPDATE`/`.update` in `SlickDataAccess.updateWorkflowState`

So, while I usually associate a Query with a Select, I don't know what to call the original compiled query variable, when it's also used for update operations.